thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/anomaly/onboard/OnboardingTaskTestUtils.java

```diff
@@ -0,0 +1,54 @@
+package com.linkedin.thirdeye.anomaly.onboard;
+
+import com.linkedin.thirdeye.anomaly.onboard.tasks.DefaultDetectionOnboardJob;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.MapConfiguration;
+
+
+public class OnboardingTaskTestUtils {
+  public static String TEST_COLLECTION = "test_dataset";
+  public static String TEST_METRIC = "test_metric";
+
+  /**
+   * Generate a default job properties for all onboarding tests
+   * @return a job properties
+   */
+  public static Map<String, String> getJobProperties(){
+    Map<String, String> properties = new HashMap<>();
+    properties.put(
+        DefaultDetectionOnboardJob.FUNCTION_FACTORY_CONFIG_PATH, ClassLoader.getSystemResource("sample-functions.properties").getPath());
+    properties.put(DefaultDetectionOnboardJob.ALERT_FILTER_FACTORY_CONFIG_PATH, ClassLoader.getSystemResource("sample-alertfilter.properties").getPath());
+    properties.put(DefaultDetectionOnboardJob.ALERT_FILTER_AUTOTUNE_FACTORY_CONFIG_PATH, ClassLoader.getSystemResource("sample-alertfilter-autotune.properties").getPath());
+    properties.put(DefaultDetectionOnboardJob.FUNCTION_NAME, "Normal Function");
+    properties.put(DefaultDetectionOnboardJob.COLLECTION_NAME, TEST_COLLECTION);
+    properties.put(DefaultDetectionOnboardJob.METRIC_NAME, TEST_METRIC);
+    properties.put(DefaultDetectionOnboardJob.WINDOW_SIZE, "1");
+    properties.put(DefaultDetectionOnboardJob.WINDOW_UNIT, "DAYS");
+    properties.put(DefaultDetectionOnboardJob.CRON_EXPRESSION, "0 0 0 1/1 * ? *");
+    properties.put(DefaultDetectionOnboardJob.FUNCTION_PROPERTIES, "metricTimezone=America/Los_Angeles;");
+    properties.put(DefaultDetectionOnboardJob.ALERT_NAME, "Normal Alert");
+    properties.put(DefaultDetectionOnboardJob.ALERT_TO, "test@test.com");
+    properties.put(DefaultDetectionOnboardJob.SMTP_HOST, "test.com");
+    properties.put(DefaultDetectionOnboardJob.SMTP_PORT, "25");
```

Comment:
Hi YungYu, thanks for doing the test for onboarding tasks. Here are a bunch of parameters that can be read from thirdeye-configs, e.g. SMTP_HOST, SMTP_PORT. Since currently onboarding tasks is taking a JSON format of payload and convert it to configurations to run, should the input payload include these parameters that are already in thirdeye-configs? Note that frontend cannot commit configurations to its codebase. 

By running the onboarding tasks end-to-end, I always failed on the "NotificationOnboardingTask" at the point when it's doing Precondition-check-not-null for parameters such as SMTP_HOST. Is this the design that frontend should pass in the parameters with regard to email sending, or we can let "NotificationOnboardingTask" to read from thirdeye-configs if it finds NULL for the configurations. Thanks!

Reply:
The parameters such as SMTP_HOST and SMTP_PORT will be injected from system configuration, which is performed at line 65-70 in DetectionOnboardResource.java.

During testing, you need to inject those parameters manually.

Reply:
I triggered the whole on-boarding using the endpoint `+  @POST
 +  @Path("/create-job")
 ` in DetectionOnboardResource.java from https://github.com/linkedin/pinot/pull/2230. I skipped the parameters that supposed to be injected from system such as SMTP_HOST / SMTP_PORT. But the whole onboarding task fails at "NotificationOnboardingTask" saying missing input. Is this a bug?